mujoco_warp/_src/ray.py

```diff
@@ -694,19 +696,39 @@ def ray(
       dist: distances from ray origins to geom surfaces
       geom_id: IDs of intersected geoms (-1 if none)
   """
-  nrays = pnt.shape[0]
+  nrays = pnt.shape[1]
   dist = wp.zeros((d.nworld, nrays), dtype=float)
   closest_hit_geom_id = wp.zeros((d.nworld, nrays), dtype=int)
-  num_threads = 64
 
   # Create default geomgroup if None is provided
   has_geomgroup = geomgroup is not None
   if geomgroup is None:
     geomgroup = vec6(0, 0, 0, 0, 0, 0)
 
+  if bodyexclude is None:
+    bodyexclude = wp.zeros(nrays, dtype=int)
```

Comment:
is this JAX-safe?  I thought we can't allocate arrays during cuda graph capture in JAX - no?

Reply:
#277 is updated to add fields `ray_dist` and `ray_geomid` to `Data` and `ray.ray` utilizes these fields instead of allocating arrays.